spec/features/trainees/filtering_trainees_spec.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.feature "Filtering trainees" do
+  before do
+    given_i_am_authenticated
+    given_trainees_exist_in_the_system
+    Trainee.update_all(provider_id: @current_user.provider.id)
```

Comment:
Can we wrap this in a method? We should try and avoid putting code directly in the `before` and `scenario` block.

Reply:
Thanks! Moved into `given_trainees_exist_in_the_system` method.